src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/ConfigurationSetting.cs

```diff
@@ -93,12 +93,18 @@ public ConfigurationSetting(string key, string value, string label = null)
         public bool Equals(ConfigurationSetting other)
         {
             if (other == null) return false;
+            if (ETag != null && other.ETag != null)
+            {
+                if (!string.Equals(ETag, other.ETag, StringComparison.Ordinal)) return false;
+                if (LastModified != other.LastModified) return false;
+            }
+            if (!string.Equals(Key, other.Key, StringComparison.Ordinal)) return false;
```

Comment:
It would be great to add a test or two that ensures that two key/values that differ in casing for the Key value (since it was not obvious to me that the key is indeed case sensitive).

Reply:
Added.
And thanks to it I caught a bug in the Clone method, so that is fixed too.